clang/lib/Driver/Driver.cpp

```diff
@@ -4638,7 +4638,10 @@ Action *Driver::BuildOffloadingActions(Compilation &C,
     }
   }
 
-  if (offloadDeviceOnly())
+  // All kinds exit now in device-only mode except for non-RDC mode HIP.
```

Comment:
I am wondering whether we should restrict this change and below change to new driver, as the old driver should not need them.

Reply:
The old driver doesn't call this function at all, they're completely separate right now.

Reply:
Basically, the new driver does everything through `BuildOffloadingActions` as a function call which augments the current job list. The old driver has a special builder class that does actions in parallel. There's a big `if (new_driver)` around both those methods.